pages/_app.tsx

```diff
@@ -1,37 +1,58 @@
 import CssBaseline from '@material-ui/core/CssBaseline'
 import { ThemeProvider } from '@material-ui/core/styles'
-import App from 'next/app'
 import Head from 'next/head'
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { blue, deepPurple } from '@material-ui/core/colors'
+import { Color } from '@material-ui/core'
+import { NextPage, NextComponentType, NextPageContext } from 'next'
+import { AppProps } from 'next/app'
 
-import theme from '../components/theme'
+import { AppContext } from '../components/AppContext'
+import { MakeTheme } from '../components/Theme'
 
-// `App` component is used for initialize of pages. It allows for overwriting and full control of the `page` initialization.
-// This allows for keeping state between navigation, custom error handling, injecting additional data.
-export default class MyApp extends App {
-	componentDidMount(): void {
-		// Remove the server-side injected CSS.
+interface Props {
+	Component: NextComponentType<NextPageContext, any, {}>
+	pageProps: any
+}
+
+const MyApp: NextPage<AppProps> = ({ Component, pageProps }: Props): JSX.Element => {
+	const [themeType, setThemeType] = useState<'dark' | 'light'>('dark')
+	const [primary, setPrimary] = useState<Color>(blue)
+	const [secondary, setSecondary] = useState<Color>(deepPurple)
+	const theme = MakeTheme(primary, secondary, themeType)
+
+	useEffect(() => {
 		const jssStyles = document.querySelector('#jss-server-side')
 		if (jssStyles) {
 			jssStyles.parentElement!.removeChild(jssStyles)
 		}
-	}
+	}, [])
 
-	render(): JSX.Element {
-		const { Component, pageProps } = this.props
-
-		return (
-			<>
-				<Head>
-					{/* https://err.sh/next.js/no-document-title */}
-					<title>My new cool app</title>
-				</Head>
-				<ThemeProvider theme={theme}>
-					{/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
-					<CssBaseline />
+	return (
+		<>
+			<Head>
+				<meta name="theme-color" content={primary[500]} />
```

Comment:
The whole point of moving meta theme-color here was so that it could use an actual value from the theme lol

Reply:
This should use the `theme` object. `primary` and `secondary` don't do anything and should be removed.